### PR TITLE
Automatic update of rewrite url only on creation

### LINKED ISSUE #25065: No slug rewrite when category or CMS rename
#### Is your feature request related to a problem?

In SEO, it is essential to never change URLs. On Prestashop, the category name or CMS page name is linked with the slug. This is a very big problem in SEO because many merchants can change the name of their categories/CMS, without knowing the impact on the URL. During my audits, in the history of the site, I regularly see URL changes, mostly on categories but also on CMS pages, only because of this feature. Sometimes even merchants who are aware of this issue do not pay attention. This has important negative impacts on search engine rankings.

Currently, after a category rename-CMS page/URL rewrite, a 301 redirect is set up automatically, but this is also impactful. It is important that the URL is never modified (or modified only if considered). And some merchants will repeatedly change the name of certain categories/CMS pages, which will result in multiple redirects.

#### Describe the solution you'd like

No link at all between category name/CMS page name and category rewrite, or after the rewrite was first written.

### PR DESCRIPTION
**Pull request by jf-viguier**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Automatic update of rewrite url only on creation to avoid SEO penalty of redirections on category and cms page
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/25065
| How to test?      | Create a new category or cms page: rewrite is updated on every name change. Update a new category or cms page: the rewrite is not updated.
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25075)

### CODE REVIEW

**Comment by jf-viguier** [comment]
If you have a better way to figure out in js if it's a creation or update page, feel free to suggest.

**Comment by Progi1984** [comment]
@jf-viguier Thanks for your contribution. I added the label "Waiting for PM" as the issue wait for @PrestaShop/product-team's answers.

**Comment by ldeprestashop** [comment]
Add "PM check" Tag : let's go :)

**Comment by jf-viguier** [comment]
Thanks @ldeprestashop

File: admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.js
Comment (jolelievre) [inline admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.js]: I understand the purpose, however it means you can never benefit of this feature when you are editing the content and you do want to update it
I would propose two solutions for this:
- ideally, we should have a button next to the rewrite-link input to force its update manually (that's the long solution)
- you could check in your condition if the input is empty, so there would be a workaround if you need to refresh it you can empty the input 🤔

Comment (jf-viguier) [reply admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.js]: 1- there is no benefit on adding a redirection, if the merchant still wants a rename, it can change it manually.
2- the rewrite can't be empty at creation time : it's mandatory

Comment (jf-viguier) [reply admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.js]: @jolelievre @ldeprestashop in product page, the button in your first case has been deleted (experimental version)
The "Force update of friendly URL" option in Product Settings is the best way IMO, so can we consider that this settings should also be relevant in category page ?

Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.js]: I'm afraid checking for `new` in URL is really not the right way to do it.
If you have `www.newstyle.fr` it will return true every time.

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
The check about `new` is not appropriate

File: admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.js
Comment (jf-viguier) [reply admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.js]: @PierreRambaud what do you suggest ? There is no hidden id in the page.

Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.js]: You can retrieve an element or create a variable to pass into javascript instead of checking the URL.

Comment (jf-viguier) [reply admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.js]: @PierreRambaud ok for that.
@PierreRambaud @ldeprestashop What do you think of using "Force update of friendly URL" option in Product Settings page ?

**Comment by Progi1984** [comment]
@jf-viguier An idea  :

In `templates/bundles/PrestaShopBundle/Admin/Sell/Catalog/Categories/edit.html.twig`, pass the `editableCategory.id` to `templates/bundles/PrestaShopBundle/Admin/Sell/Catalog/Categories/Blocks/form.html.twig` and add it to 
`{{ form_start(categoryForm, {'attr': {'id': categoryId}}) }}` and you have the solution to check if it is a solution or not.

**Comment by jf-viguier** [comment]
@Progi1984 thanks for the help on twig but I've used SF controller instead of twig forms.
My js code is now clean et works for category and cms page.

**Review by Progi1984 — DISMISSED** [review]

**Comment by jf-viguier** [comment]
@Progi1984 I've rebased

File: admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.ts
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.ts]:
```suggestion
const {$} = window;
```

File: admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.ts
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.ts]:
```suggestion
  options = {eventName: 'input'},
```

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

File: src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php]: Just a suggestion but you could get rid of this in the controller, and in the twig template use `'data-id': cmsPageId|default(0)`

Just to keep the controller cleaner and to make sure the form rendering is resillient

Comment (jf-viguier) [reply src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php]: Thanks @jolelievre, it's done. I've no time to lean twig for now but I'll do it.

**Review by Progi1984 — DISMISSED** [review]

File: templates/bundles/PrestaShopBundle/Admin/Improve/Design/Cms/Blocks/form.html.twig
Comment (PierreRambaud) [inline templates/bundles/PrestaShopBundle/Admin/Improve/Design/Cms/Blocks/form.html.twig]:
```suggestion
{{ form_start(cmsPageForm, {'attr': {'data-object-id': cmsPageId|default(0)}}) }}
```

File: templates/bundles/PrestaShopBundle/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (PierreRambaud) [inline templates/bundles/PrestaShopBundle/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]:
```suggestion
{{ form_start(categoryForm, {'attr': {'data-object-id': categoryId|default(0)}}) }}
```

it's not a form id, the id is related to the object. We have to specify it in the name

**Comment by jf-viguier** [comment]
I've commited your suggestions on data-object-id stuff @PierreRambaud thanks

File: admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.ts
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.ts]:
```suggestion
    if (!$(event.currentTarget).closest('form').data('objectId')) {
```
?

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
@jf-viguier I just approved your PR. And I didn't see the conflicting files, could you rebase, please ?

**Comment by matks** [comment]
Hi @jf-viguier ,

Since we had no news from you for more than 30 days, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale. It does not mean we dont approve the changes, it's just to keep our backlog of PRs manageable.

Naturally if you want to come back on this PR, rebased, and reopen it I'll be happy to review it 🎉 .

**Comment by jf-viguier** [comment]
hi @matks @Progi1984, I've just rebased, can you please reopen this PR ?

**Comment by matks** [comment]
Weird 🤔  I cant, github tells me this
Capture d’écran 2021-12-06 à 11 35 31

Maybe we have to open a new PR

**Comment by jf-viguier** [comment]
@matks thanks, I've opened the new pr here https://github.com/PrestaShop/PrestaShop/pull/26890

**Pull request closed without merging**
